lib/ansible/executor/process/threading.py

```diff
@@ -0,0 +1,171 @@
+# (c) 2018, Red Hat, Inc. <support@ansible.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+import sys
+import threading
+import time
+import traceback
+
+from collections import deque
+from jinja2.exceptions import TemplateNotFound
+
+from ansible.errors import AnsibleConnectionFailure
+from ansible.executor.process.model import ProcessModelBase, ResultsSentinel, keyboard_interrupt_event
+from ansible.executor.task_executor import TaskExecutor
+from ansible.executor.task_result import TaskResult
+from ansible.module_utils._text import to_text
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+__all__ = ['ProcessModelThreading', ]
+
+
+class ProcessModelThreading(ProcessModelBase):
+    def __init__(self, tqm):
+        super(ProcessModelThreading, self).__init__(tqm)
+
+        self._job_queue = deque()
+        self._job_queue_lock = threading.Lock()
+
+    def initialize_workers(self, num):
+        self._cur_worker = 0
+        self._workers = []
+        for i in range(num):
+            w_thread = threading.Thread(target=run_worker, args=(self, self._shared_loader_obj))
+            w_thread.start()
+            w_lock = threading.Lock()
```

Comment:
This lock is not used anywhere, and we removed a similar lock from forking, so we need to remove it here too.